logbook-spring-boot-starter/src/main/java/org/zalando/logbook/spring/LogbookAutoConfiguration.java

```diff
@@ -0,0 +1,184 @@
+        registration.setName(UNAUTHORIZED);
+        registration.setDispatcherTypes(REQUEST, ASYNC, ERROR);
+        registration.setOrder(Ordered.HIGHEST_PRECEDENCE + 1);
+        return registration;
+    }
+
+    @Bean
+    @ConditionalOnWebApplication
+    @ConditionalOnProperty(name = "logbook.filter.enabled", havingValue = "true", matchIfMissing = true)
+    @ConditionalOnMissingBean(name = AUTHORIZED)
+    public FilterRegistrationBean authorizedLogbookFilter(final Logbook logbook) {
+        final Filter filter = new LogbookFilter(logbook);
+        final FilterRegistrationBean registration = new FilterRegistrationBean(filter);
+        registration.setName(AUTHORIZED);
+        registration.setDispatcherTypes(REQUEST, ASYNC, ERROR);
+        registration.setOrder(Ordered.LOWEST_PRECEDENCE);
+        return registration;
+    }
+
+    @Bean
+    @ConditionalOnMissingBean(Logbook.class)
+    public Logbook logbook(final Obfuscator headerObfuscator,
+            final Obfuscator parameterObfuscator,
+            final BodyObfuscator bodyObfuscator,
+            @SuppressWarnings("SpringJavaAutowiringInspection") final HttpLogFormatter formatter,
+            final HttpLogWriter writer) {
+        return Logbook.builder()
+                .headerObfuscator(headerObfuscator)
+                .parameterObfuscator(parameterObfuscator)
+                .bodyObfuscator(bodyObfuscator)
+                .formatter(formatter)
+                .writer(writer)
+                .build();
+    }
+
+    @Bean
+    @ConditionalOnMissingBean(name = "headerObfuscator")
+    public Obfuscator headerObfuscator() {
+        final List<String> headers = properties.getObfuscate().getHeaders();
+        return headers.isEmpty() ?
+                authorization() :
+                createObfuscator(headers, String::equalsIgnoreCase);
+    }
+
+    @Bean
+    @ConditionalOnMissingBean(name = "parameterObfuscator")
+    public Obfuscator parameterObfuscator() {
+        return createObfuscator(properties.getObfuscate().getParameters(), String::equals);
+    }
+
+    private Obfuscator createObfuscator(final Collection<String> names, final BiPredicate<String, String> matcher) {
+        return compound(names.stream()
+                .map(name -> obfuscate(actual ->
+                        matcher.test(name, actual), "XXX"))
+                .collect(toList()));
+    }
+
+    @Bean
+    @ConditionalOnMissingBean(BodyObfuscator.class)
+    public BodyObfuscator bodyObfuscator() {
+        return BodyObfuscator.none();
+    }
+
+    @Bean
+    @ConditionalOnMissingBean(HttpLogFormatter.class)
+    @ConditionalOnProperty(name = "logbook.format.style", havingValue = "http")
+    public HttpLogFormatter httpFormatter() {
+        return new DefaultHttpLogFormatter();
+    }
+
+    @Bean
+    @ConditionalOnBean(ObjectMapper.class)
+    @ConditionalOnMissingBean(HttpLogFormatter.class)
+    public HttpLogFormatter jsonFormatter(
```

Comment:
> What happens if you have Jackson on the classpath and style set to http ?

I checked and my current test is configuring the objectmapper in both tests, but as son as we specify http style, we get the http formatter.